compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -305,17 +296,50 @@ static LogicalResult checkWhetherSplitIsPossible(
   return success();
 }
 
-// Given a vector of L2->L1 Dma ops' perform the splitting :-
-// 1. Check if the splitting can be performed or not. If not possible, bail out.
-// 2. For the split dimension inferred set offset = 0 and size as 1 for L2 and
-//    L3.
-// 3. Now traverse each L2->L1 Dma op and perform the following :-
-//    a) Create a new L2 AllocOp based on the updated size (step 3 above) and
-//       create a logicalobjectfifo using the same.
-//    b) Split L3->L2 Dma op.
-//    c) SPlit L2->L1 Dma op.
-// 4. Delete old L2->L1, L3->L2 and corresponding AllocOps.
-LogicalResult splitLogicalObjectFifos(
+/// Utility to check if the L3->L2 dma is transposed on the L2 side.
+/// The intuition is if the sizes of L2 and L3 side are the same and L2 dma
+/// strides are in non-decreasing order from right to left, then the dma has
+/// transposition on L3 side, otherwise the transposition is on L2 side.
+static FailureOr<bool> isL2DmaTransposed(AMDAIE::DmaCpyNdOp l3ToL2DmaOp,
+                                         bool isL2Target) {
+  SmallVector<OpFoldResult> l2DmaSizes;
+  SmallVector<OpFoldResult> l2DmaStrides;
+  SmallVector<OpFoldResult> l3DmaSizes;
+  bool l2DmaTransposed = true;
```

Comment:
I suggest moving this closer to the use. I.e. before https://github.com/nod-ai/iree-amd-aie/pull/910/files#diff-591f54662e604611ca835020aa99f74410d5e332fda12400ccf2eca421d2283bR325